src/pp-annotator/form/AnnotationForm.tsx

```diff
@@ -123,16 +110,8 @@ export default class AnnotationForm extends React.Component<
     this.setState({ [name]: target.value });
   }
 
-  private handleIsLinkOnlyChange(e: React.ChangeEvent<HTMLInputElement>) {
-    this.setState({ isLinkOnly: e.currentTarget.checked });
-  }
-
   private onSave(event: any) {
-    const fieldsToSave = getFormState(this.state);
-    if (this.state.isLinkOnly) {
-      fieldsToSave.comment = '';
-    }
-    this.props.annotation.fields = fieldsToSave;
+    Object.assign(this.props.annotation, getFormState(this.state));
```

Comment:
hmm, nie rozumiem za bardzo zamysłu tej linijki?

Reply:
Przepisuje odpowiednie pola z formularza na `AnnotationViewModel` (co jest potrzebne, bo w ViewModelu są zapisane też inne dane, o których komponent nie ma pojęcia). Dopiero potem robi.
```js
this.props.saveAction(this.props.annotation);
```

Reply:
spoko, ale IMO lepiej byłoby tworzyć nowy obiekt i do niego wstawiać wartości z propsów i z state.
W tej chwili modyfikowane są propsy, a to jest w Reactcie "zakazane" :P